test_socket.py

```diff
@@ -0,0 +1,232 @@
+import socket
+import string
+import email.utils
+
+
+def test_text():
+    ADDR = ('127.0.0.1', 10007)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    msg = "GET sample.txt HTTP/1.1"+"\r\n"+"Host: www.example.com"+"\r\n\r\n"
```

Comment:
Ahhhhhh, I begin to see.

When a browser generates a request, it always makes sure that the path starts with a slash.  Your paths do not start with slashes which is why your tests work, even though you are joining the URI to the ROOT with os.path.join.  Fix your tests by adding a slash at the start of every URI and you should see them fail with a 404 in all cases.  Then fix your server to remove the slash at the start of URI.  This will be more realistic.